### PR TITLE
update: skip scroll to top on inital SSR completion.

### LINKED ISSUE #6518: Page scrolls to top after initial load.
If I start to load a page (initial load via SSR) and start to scroll down, the page scrolls back to the top when it is done loading, which interrupts me. I guess this comes from the ScrollToTop component connected to the router. This is okay for navigating after the initial load (with some exceptions like #3662) but we should try to avoid it for the initial load.

### PR DESCRIPTION
**Pull request by Faakhir30**
- [x] I signed and returned the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement), and received and accepted an invitation to join a team in the Plone GitHub organization.
- [x] I verified there aren't other open [pull requests](https://github.com/plone/volto/pulls) for the same change.
- [x] I followed the guidelines in [Contributing to Volto](https://6.docs.plone.org/volto/contributing/index.html).
- [x] I succesfully ran [code linting checks](https://6.docs.plone.org/volto/contributing/linting.html) on my changes locally.
- [x] I succesfully ran [unit tests](https://6.docs.plone.org/volto/contributing/testing.html) on my changes locally.
- [x] I succesfully ran [acceptance tests](https://6.docs.plone.org/volto/contributing/acceptance-tests.html) on my changes locally.
- [x] If needed, I added new tests for my changes.
- [x] If needed, I added [documentation](https://6.docs.plone.org/volto/contributing/documentation.html#narrative-documentation) for my changes, either in the Storybook or narrative documentation.
- [x] I included a [change log entry](https://6.docs.plone.org/contributing/index.html#contributing-change-log-label) in my commits.

-----

Closes #6518 , skipping scroll to top on first `componenetDidUpdate` after SSR, as that is being triggered by SSR completion and is causing that intruption

### CODE REVIEW

**Review by stevepiercy — APPROVED** [review]
News looks good to me.

Needs a technical review from @plone/volto-accessibility and @plone/volto-team.

Thank you!

File: packages/volto/news/6523.internal
Comment (stevepiercy) [inline packages/volto/news/6523.internal:1]: Whoops, reviewed too quickly. This file should be renamed to `6523.bugfix`.

**Review by stevepiercy — CHANGES_REQUESTED** [review]

File: packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx [hunk 1]
```diff
@@ -22,12 +28,30 @@ class ScrollToTop extends React.Component {
     children: PropTypes.node.isRequired,
   };
 
+  /**
+   * Used to indicate client-side rendering
+   * @memberof ScrollToTop
+   */
+  componentDidMount() {
+    this.isClientSide = true;
```
Comment (avoinea) [inline packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx:36]: @Faakhir30 I think the convention for this variable name is `isClient` and maybe you should use the component state for it :thinking: 

See for example: https://github.com/plone/volto/blob/main/packages/volto/src/components/theme/View/View.jsx#L114

**Review by avoinea — CHANGES_REQUESTED** [review]

File: packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx (hunk 1 above)
Comment (Faakhir30) [reply packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx]: @avoinea Thanks for sharing the sample reference. I did think, IMO, that `isClient` wasn't required in my case, so I removed that variable in my recent commit. The reason is that `componentDidUpdate` never gets triggered during SSR; it is only invoked on the client side.

File: packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx [hunk 2]
```diff
@@ -22,12 +22,21 @@ class ScrollToTop extends React.Component {
...
    * @returns {null} Null
    * @memberof ScrollToTop
    */
   componentDidUpdate(prevProps) {
+    // Skip the first client-side update that happens right after hydration
+    if (this.state.isFirstClientUpdatePending) {
+      this.setState({ isFirstClientUpdatePending: false });
+      return;
+    }
+
```
Comment (davisagli) [inline packages/volto/src/helpers/ScrollToTop/ScrollToTop.jsx:39]: @Faakhir30 We can do this a bit more simply without using component state. Instead, add a check for `&& prevProps.location !== undefined` after line 41. Could you make this change please?

**Review by davisagli — CHANGES_REQUESTED** [review]

**Review by davisagli — APPROVED** [review]
Thanks, looks good to me now.

**Pull request merged**